File: iOS/TUILiveKit/Source/State/View/ViewState.swift
//
//  VoiceRoomNavigationState.swift
//  VoiceRoom
//
//  Created by aby on 2024/3/1.
//

import Foundation

struct ViewState {
    var linkStatus: LinkStatus = .none
    var liveStatus: LiveStatus = .previewing
    var autoOpenCameraOnSeated: Bool = true
}

File: Flutter/livekit/lib/live_stream/state/view/view_state.dart
import 'package:flutter/material.dart';

import '../index.dart';

class ViewState {
  final ValueNotifier<LinkStatus> linkStatus = ValueNotifier<LinkStatus>(LinkStatus.none);
  final ValueNotifier<bool> autoOpenCameraOnSeated = ValueNotifier<bool>(false);
  final ValueNotifier<LiveStatus> liveStatus = ValueNotifier<LiveStatus>(LiveStatus.none);

  void reset() {
    linkStatus.value = LinkStatus.none;
    autoOpenCameraOnSeated.value = false;
    liveStatus.value = LiveStatus.none;
  }
}
